sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1091,8 +1091,13 @@ abstract class ToTimestamp
   override protected def formatString: Expression = right
   override protected def isParsing = true
 
+  override def forTimestampNTZ: Boolean = left.dataType match {
+    case TimestampNTZType => true
+    case _ => false
+  }
+
```

Comment:
Would be simpler to invoke isInstanceOf? Like
```scala
override def forTimestampNTZ: Boolean = left.dataType.isInstanceOf[TimestampNTZType]
```

Reply:
@beliefer Could you re-trigger GAs since they have more tests than jenkins build. And the method can be rewritten in simpler way I guess:
```scala
  override val forTimestampNTZ: Boolean = left.dataType == TimestampNTZType
```